python/source/io.rst

```diff
@@ -419,7 +422,8 @@ by ``month`` using
 
 .. testcode::
 
-    dataset = ds.dataset("s3://ursa-labs-taxi-data/2011", 
+    dataset = ds.dataset("2011", 
+                         filesystem=s3
```

Comment:
Hmm, I suppose I feel this changes the intent a little if the goal was to demonstrate the ability to create a dataset from a URI.

That being said, I don't know how to make this test work effectively with a URI.  If `anonymous` is not provided then credentials will be fetched from by the AWS SDK (e.g. from ~/.aws/credentials).  If no credentials at all are found then the request will fail.  If you configure any kind of credentials then the request will pass.  Perhaps the C++ library could catch the "no credentials found" error and try again with anonymous access.

Reply:
@westonpace I understand the reasoning about the URI. Perhaps, instead of changing this to `filesystem` we do, as you suggest, add `~/.aws/credentials` as an explicit step before this code? I think this would make sure this code works even if credentials are not present.

Reply:
@westonpace I changed the code back to using URI and added a warning that if this code fails most likely it's because of the AWS credentials not being set. Added instructions how to set these properly.

Reply:
Thanks, I'm good with this approach.  If amol- or jorisvandenbossche have better ideas when they get back we can fix it then.